kolibri/plugins/coach/assets/src/views/plan/LessonEditDetailsPage/index.vue

```diff
@@ -106,13 +108,58 @@
         }
         return this.$router.getRoute(route);
       },
+      initialAdHocLearners() {
+        return this.$store.state.adHocLearners.user_ids;
+      },
+      initialSelectedCollectionIds() {
+        let collectionIds = [];
+        // Only include the AdHocGroup in this if it has already
+        // had learners assigned to it.
+        this.lesson.lesson_assignments.forEach(assignment => {
+          if (assignment.collection_kind === CollectionKinds.ADHOCLEARNERSGROUP) {
+            if (this.hasAdHocLearnersAssigned) {
+              collectionIds.push(assignment.collection);
+            }
+          } else {
+            collectionIds.push(assignment.collection);
+          }
```

Comment:
A couple thoughts and questions here, nothing critical though:
- I think this logic could be simplified
- I'm not sure, but perhaps a `.filter` and `.map` would be slightly simpler
- Is it possible that this could return a set of ID's that isn't unique? Would that matter?